stride/problem/acquisitions.py

```diff
@@ -979,6 +980,77 @@ def default(self):
                           geometry=self._geometry, problem=self.problem))
 
+    def from_fullwave(self, acquisition_path, source_path=None, read_traces=False):
+        """
+        Populates acquisition container with shot and receiver ids as described
+        by a Fullwave .ttr acquisition file and an optional source .ttr file
+        describing the wavelets for each shot. Assumes point-source scheme, i.e.,
+        that each source id is an individual shot. Composite sources are yet to be
+        added to functionality
+
+        Parameters
+        ----------
+        acquisition_path : str
+            Path to .ttr acquisition Fullwave file (ideally Observed-Trace.ttr)
+        source_path: str, optional
+            Path to .ttr or .txt source Fullwave file. Default None
+        read_traces: bool, optional
+            If flagged, data from acquisition_path file is read and for each source id
+            and added to its correspondent Stride ``Shot`` object. Default False.
+
+        Returns
+        -------
+
+        """
+
+        assert acquisition_path.lower().split(".")[-1] == "ttr", "Expected .ttr extension in\
+             acquisition_path but found .%s " % acquisition_path.lower().split(".")[-1]
+        if source_path is not None:
+            assert source_path.lower().split(".")[-1] in ("ttr", "txt"), "Expected .ttr or .\
+                txt extension in source_path but found .%s" % source_path.lower().split(".")[-1]
+
+        # Import of fullwave module at the top is throwing circular import error, needs debugging
+        from ..utils.fullwave import read_observed_ttr, read_signature_ttr, read_signature_txt
+
+        # Read acquisition file
+        sources_ids, receiver_ids, shottraces = read_observed_ttr(acquisition_path, read_traces)
+
+        # Check every source has an assigned list of receivers and a trace
+        assert len(sources_ids) == len(receiver_ids), (len(sources_ids), len(receiver_ids))
+        if len(shottraces) > 1:
+            assert len(shottraces) == len(sources_ids), (len(shottraces), len(sources_ids))
+
+        # Read source signature file
+        if source_path is not None:
+            srcext = source_path.lower().split(".")[-1]
+            if srcext == "ttr":
+                wavelets = read_signature_ttr(source_path)
+
+            elif srcext == "txt":
+                wavelet = read_signature_txt(source_path)
+                wavelet = np.array(wavelet, dtype=np.float32)
+                wavelets = np.broadcast_to(wavelet, (len(sources_ids), len(wavelet)))
+
+        # Add each source and correspondent receivers as a shot
+        for i, sid in enumerate(sources_ids):
+            source = self._geometry.get(sid)
+            receivers = [self._geometry.get(rid) for rid in receiver_ids[i]]
+            shot = Shot(sid,
+                          sources=[source], receivers=receivers,
+                          geometry=self._geometry, problem=self.problem)
```

Comment:
Please, could we align the indentation with sid?

Reply:
See commit [4561726](https://github.com/trustimaging/stride/pull/30/commits/4561726a69d476df160dfcfe4964b51b81859c37)